pkg/controller/direct/bigquerydataset/bigquerydataset_mappings.go

```diff
@@ -21,314 +21,219 @@
-	if in == nil {
-		return nil
-	}
-	out := &pb.ExternalDatasetReference{}
-	out.ExternalSource = direct.ValueOf(in.ExternalSource)
-	out.Connection = direct.ValueOf(in.Connection)
-	return out
-}
-func ListFormatDataset_FromProto(mapCtx *direct.MapContext, in *pb.Dataset) *krm.ListFormatDataset {
-	if in == nil {
-		return nil
+	out := &krm.EncryptionConfiguration{}
+	out.KmsKeyRef = &v1beta1.KMSCryptoKeyRef{
+		External: in.KmsKeyName,
 	}
-	out := &krm.ListFormatDataset{}
-	out.Kind = direct.LazyPtr("BigQueryDataset")
-	out.DatasetReference = DatasetReference_FromProto(mapCtx, in)
 	return out
 }
-func ListFormatDataset_ToProto(mapCtx *direct.MapContext, in *krm.ListFormatDataset) *pb.Dataset {
+func RoutineReference_FromProto(mapCtx *direct.MapContext, in *api.RoutineReference) *krm.RoutineReference {
 	if in == nil {
 		return nil
 	}
-	out := &pb.Dataset{}
-	out = DatasetReference_ToProto(mapCtx, in.DatasetReference)
+	out := &krm.RoutineReference{}
+	out.DatasetId = direct.LazyPtr(in.DatasetId)
+	out.ProjectId = direct.LazyPtr(in.ProjectId)
+	out.RoutineId = direct.LazyPtr(in.RoutineId)
 	return out
 }
-func RoutineReference_FromProto(mapCtx *direct.MapContext, in *pb.Routine) *krm.RoutineReference {
+func RoutineReference_ToProto(mapCtx *direct.MapContext, in *krm.RoutineReference) *api.RoutineReference {
 	if in == nil {
 		return nil
 	}
-	out := &krm.RoutineReference{}
-	out.DatasetId = &in.DatasetID
-	out.ProjectId = &in.ProjectID
-	out.RoutineId = &in.RoutineID
+	out := &api.RoutineReference{}
+	out.DatasetId = direct.ValueOf(in.DatasetId)
+	out.ProjectId = direct.ValueOf(in.ProjectId)
+	out.RoutineId = direct.ValueOf(in.RoutineId)
 	return out
 }
-func RoutineReference_ToProto(mapCtx *direct.MapContext, in *krm.RoutineReference) *pb.Routine {
+func TableReference_FromProto(mapCtx *direct.MapContext, in *api.TableReference) *krm.TableReference {
 	if in == nil {
 		return nil
 	}
-	out := &pb.Routine{}
-	out.DatasetID = *in.DatasetId
-	out.ProjectID = *in.ProjectId
-	out.RoutineID = *in.RoutineId
+	out := &krm.TableReference{}
+	out.DatasetId = direct.LazyPtr(in.DatasetId)
+	out.ProjectId = direct.LazyPtr(in.ProjectId)
+	out.TableId = direct.LazyPtr(in.TableId)
 	return out
 }
-func TableReference_FromProto(mapCtx *direct.MapContext, in *pb.Table) *krm.TableReference {
+func TableReference_ToProto(mapCtx *direct.MapContext, in *krm.TableReference) *api.TableReference {
 	if in == nil {
 		return nil
 	}
-	out := &krm.TableReference{}
-	out.DatasetId = &in.DatasetID
-	out.ProjectId = &in.ProjectID
-	out.TableId = &in.TableID
+	out := &api.TableReference{}
+	out.DatasetId = direct.ValueOf(in.DatasetId)
+	out.ProjectId = direct.ValueOf(in.ProjectId)
+	out.TableId = direct.ValueOf(in.TableId)
 	return out
 }
-func TableReference_ToProto(mapCtx *direct.MapContext, in *krm.TableReference) *pb.Table {
+func BigQueryDataset_CovertKRMToAPI(mapCtx *direct.MapContext, in *krm.BigQueryDataset) *api.Dataset {
```

Comment:
This function is probably not needed. We only need to send spec fields (i.e. fields that can be configured by users) to GCP when creating the resource.